static/src/js/components/Map/Map.jsx

```diff
@@ -214,7 +219,7 @@ const Map = ({
     })
 
     return () => map.setTarget(null)
-  }, [projectionCode])
+  }, [projectionCode, isFocusedCollectionPage, colorMap])
```

Comment:
Instead of creating a whole new map when changing pages, create a new useEffect based on isFocusedCollectionPage and colorMap. Within that use effect you can add and remove controls based on those values.

It might use commands like this (I haven't tested it)
```
    map.addControl(new LegendControl())

    map.removeControl(map.getControls().getArray().find((control) => control instanceof LegendControl))
```

If that works, I wonder if you still need to have the `else` block in the `render` method of LegendControl